druid/examples/anim.rs

```diff
@@ -12,52 +12,62 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-//! An example of an animating widget.
+//! An example of an animating widget. It is just a widget that
+//! requests an animation frame when it needs to, and draws the frame in the
+//! `paint` method.
+//! Once the animation is over it simply stops requesting animation frames.
+//! In this case the animation time(t) is in the data so that other widgets
+//! can also read the state of the animation if needed. This can also be in
+//! the widget itself, so that it is not part of the data.
 
 use std::f64::consts::PI;
 
 use druid::kurbo::{Circle, Line};
 use druid::widget::prelude::*;
 use druid::{AppLauncher, Color, LocalizedString, Point, Vec2, WindowDesc};
 
-struct AnimWidget {
-    t: f64,
-}
+struct AnimWidget {}
```

Comment:
I think having the time in the data makes sense for this tiny example, but I think it's a better practice in larger apps to have the animation state in the widget, not the data. And then because people look at the examples to get an idea of best practices, I think it's better to have it in the widget here.

The reason animation state usually lives in the widget is because it's a piece of transient data that isn't usually interesting for the app as a whole. You can see the built-in `Switch` widget for an example -- it has a `bool` related to the animation state, and it would be annoying if every time you used `Switch`, you had to reserve a `bool` for it in your app's data.

Reply:
That makes sense, thanks for letting me know. Maybe something for in the book to talk about as well?
I can open a issue about this if that's the way to go about this.

Reply:
that would be good, this is something we've discussed in the past, and it we should write it down.

Reply:
#1337